Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -264,71 +264,123 @@ ThrowCompletionOr<Value> calendar_day_of_week(VM& vm, Object& calendar, Object&
+    if (result.is_undefined())
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.daysInMonth.as_string(), vm.names.undefined.as_string());
+
+    // 3. Return ? ToPositiveInteger(result).
+    return TRY(to_positive_integer(vm, result));
 }
 
 // 12.2.17 CalendarDaysInYear ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendardaysinyear
 ThrowCompletionOr<Value> calendar_days_in_year(VM& vm, Object& calendar, Object& date_like)
 {
     // 1. Assert: Type(calendar) is Object.
 
-    // 2. Return ? Invoke(calendar, "daysInYear", « dateLike »).
-    return TRY(Value(&calendar).invoke(vm, vm.names.daysInYear, &date_like));
+    // 2. Let result be ? Invoke(calendar, "daysInYear", « dateLike »).
+    auto result = TRY(Value(&calendar).invoke(vm, vm.names.daysInYear, &date_like));
+
+    // NOTE: Explicitly handled for a better error message similar to the other calendar property AOs
+    if (result.is_undefined())
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.daysInYear.as_string(), vm.names.undefined.as_string());
+
+    // 3. Return ? ToPositiveInteger(result).
+    return TRY(to_positive_integer(vm, result));
 }
 
 // 12.2.18 CalendarMonthsInYear ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmonthsinyear
 ThrowCompletionOr<Value> calendar_months_in_year(VM& vm, Object& calendar, Object& date_like)
 {
     // 1. Assert: Type(calendar) is Object.
 
-    // 2. Return ? Invoke(calendar, "monthsInYear", « dateLike »).
-    return TRY(Value(&calendar).invoke(vm, vm.names.monthsInYear, &date_like));
+    // 2. Let result be ? Invoke(calendar, "monthsInYear", « dateLike »).
+    auto result = TRY(Value(&calendar).invoke(vm, vm.names.monthsInYear, &date_like));
+
+    // NOTE: Explicitly handled for a better error message similar to the other calendar property AOs
+    if (result.is_undefined())
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.monthsInYear.as_string(), vm.names.undefined.as_string());
+
+    // 3. Return ? ToPositiveInteger(result).
+    return TRY(to_positive_integer(vm, result));
 }
 
 // 12.2.29 CalendarInLeapYear ( calendar, dateLike ), https://tc39.es/proposal-temporal/#sec-temporal-calendarinleapyear
 ThrowCompletionOr<Value> calendar_in_leap_year(VM& vm, Object& calendar, Object& date_like)
 {
     // 1. Assert: Type(calendar) is Object.
 
-    // 2. Return ? Invoke(calendar, "inLeapYear", « dateLike »).
-    return TRY(Value(&calendar).invoke(vm, vm.names.inLeapYear, &date_like));
+    // 2. Let result be ? Invoke(calendar, "inLeapYear", « dateLike »).
+    auto result = TRY(Value(&calendar).invoke(vm, vm.names.inLeapYear, &date_like));
+
+    // 3. Return ToBoolean(result).
+    return result.to_boolean();
```

Comment:
Just making sure that it is intentional that this doesn't have the explicit error check?
(I guess yes because to_boolean doesn't throw)

Reply:
Correct :)